web/app/scripts/state/recordstate-service.js

```diff
@@ -20,6 +27,8 @@
          */
         function init() {
           selected = null;
+          gettingSelected = false;
+          gettingOptions = false;
```

Comment:
Just curious: why are these variables needed in this service, but not the other services?